chainer/functions/connection/deconvolution_2d.py

```diff
@@ -41,11 +41,12 @@ def __init__(self, stride=1, pad=0, outsize=None, **kwargs):
             "the gradient w.r.t. x is automatically decided during "
             "backpropagation."
         )
-        argument.assert_kwargs_empty(kwargs)
```

Comment:
Do not remove this statement.
Please move to after `parse_kwargs`.

Reply:
Should `assert_kwargs_empty` be called after `parse_kwargs`? It looks it is checked at `parse_kwargs` whether `kwargs` is empty or not.

Reply:
`parse_kwargs` uses `pop`.
maybe `assert_kwargs_empty` works correctly.
Cloud you try it?

Reply:
I've just confirmed that unexpected arguments can be detected by `parse_kwargs()`. It looks, if there is `parse_kwargs()`, it is not necessary to call `assert_kwargs_empty()`.  Followings are error message when a unexpected argument (in this case, dummy) is added to `convolution_2d`.

      File "/home/anaruse/.pyenv/versions/3.5.1/lib/python3.5/site-packages/chainer-3.0.0rc1-py3.5.egg/chainer/functions/connection/convolution_2d.py", line 442, in convolution_2d
        dilate, = argument.parse_kwargs(kwargs, ('dilate', 1))
      File "/home/anaruse/.pyenv/versions/3.5.1/lib/python3.5/site-packages/chainer-3.0.0rc1-py3.5.egg/chainer/utils/argument.py", line 12, in parse_kwargs
        raise TypeError('got unexpected keyword argument(s) %s' % args)
    TypeError: got unexpected keyword argument(s) 'dummy'

Reply:
OK, I understood.